### DISCUSSION TITLE
Help us test Restforce v6.0.0, with support for Faraday v2.x

### DISCUSSION THREAD

**Discussion by timrogers** [restforce/restforce#733] [Announcements]
Restforce uses a gem called [`faraday`](https://github.com/lostisland/faraday) to make HTTP requests to the Salesforce API. 

Today, Restforce supports Faraday versions between v0.9.0 and v1.10.0. This stops you from using current, modern v2.x versions of Faraday.

This means that you can't benefit from Faraday's latest features and security fixes, and it may stop you upgrading other gems you use which require Faraday v2.0.0 or later. This isn't great.

To move Restforce forward, we've released a new, experimental version, [`v6.0.0.rc.1`](https://github.com/restforce/restforce/releases/tag/v6.0.0.rc.1).  You'll find the code in the [`v6`](https://github.com/restforce/restforce/tree/v6) branch. This version:

* Adds support for `faraday` versions `v2.0.0` onwards
* ⚠️  Drops compatability with `faraday` versions before `v1.1.0`
* ⚠️  Drops support for Ruby 2.6, since [Ruby 2.6 has reached its end-of-life](https://www.ruby-lang.org/en/downloads/)

Clearly, dropping support for Faraday v0.x and the v1.0.x minor version is painful, but I do believe that it's the right thing to do. There are a lot of differences between the Faraday versions, and for an open-source project with one active maintainer, it's a tonne of work and cognitive load to manage all of that. It increases the complexity of the code, and makes bugs much more likely.

**We're looking for Restforce users to help us test this new version, [`v6.0.0.rc.1`](https://github.com/restforce/restforce/releases/tag/v6.0.0.rc.1). All you need to do is point your project to use `v6.0.0.rc.1` in your `Gemfile`, and let us know how you get on:**

* Do your tests still pass?
* Does everything work as expected in development?
* Does everything work as expected in production?
* If you stick with Faraday v1.x, does everything still work?
* If you choose to upgrade to Faraday v2.x, does all of that go smoothly?

If you have other code that depends on Faraday, you may be forced to update that code to support a later Faraday version - see our [`UPGRADING.md`](https://github.com/restforce/restforce/blob/v6/UPGRADING.md) file for help with that.

Once we've had some initial feedback to say that everything is working as expected, we will release a final `v6.0.0` version.

Thanks for using Restforce ✨ 🚢 ❤️

**Comment by timrogers** [comment]
@thatguysimon @albandum @mcfoton 👋🏻 Hi there! Did any of you get a chance to test this out? I'd love to be able to do a final release.

**Reply by thatguysimon** [reply]
Hi @timrogers, the codebase I was working on only needed restforce for a one-time migration script so I'm afraid I won't be of much help.
Thanks for the awesome work on this gem though!

**Reply by albandum** [reply]
Hi, 
I've mostly been using the .query method and the .picklist_values one, and so far so good ! Good job :)

**Comment by timrogers** [comment]
v6.0.0 has now been released 🥳
